pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/ElasticSearchSink.java

```diff
@@ -120,22 +123,21 @@ public void write(Record<GenericObject> record) throws Exception {
                     switch (elasticSearchConfig.getNullValueAction()) {
                         case DELETE:
                             if (idAndDoc.getLeft() != null) {
+                                metrics.incrementCounter(ElasticSearchMetrics.DELETE, 1);
```

Comment:
`bulkDelete` or `deleteDocument` can fail and increment failure, but you increase it anyway- double counting.

Reply:
@asafm I changed to DELETE_ATTEMPT to count the number of attempts.

I'm having a problem to separate DELETE or index in the bulk operation. This requires more code change. Can I do this in another PR? We do need a delete_attempt count that helps to track if or how many delete documents requests.

Reply:
Ok. Ping me in the PR.

Reply:
Don't you prefer to first fix the code to have everything you need and only then add the metrics?